tests/iio_common.c

```diff
@@ -207,22 +242,32 @@ static const char *common_options_descriptions[] = {
 };
 
 
-struct iio_context * handle_common_opts(char * name, int argc, char * const argv[],
+struct iio_context * handle_common_opts(char * name, int argc,
+		char * const argv[], const char *optstring,
 		const struct option *options, const char *options_descriptions[])
 {
 	struct iio_context *ctx = NULL;
-	int c, option_index = 0;
+	int c;
 	enum backend backend = IIO_LOCAL;
 	const char *arg = NULL;
 	bool do_scan = false, detect_context = false;
+	char buf[128];
+	struct option *opts;
 
 	/* Setting opterr to zero disables error messages from getopt_long */
 	opterr = 0;
 	/* start over at first index */
 	optind = 1;
 
-	while ((c = getopt_long(argc, argv, COMMON_OPTIONS,	/* Flawfinder: ignore */
-			common_options, &option_index)) != -1) {
+	iio_snprintf(buf, sizeof(buf), "%s%s", COMMON_OPTIONS, optstring);
+	opts = add_common_options(options);
+	if (!opts) {
+		fprintf(stderr, "Memory error\n");
```

Comment:
To me, it makes more sense to say here: "failed to add common options" and inside add_common_options to add the message "Memory error" after calloc() fails. Today add_common_option() might fail due to one reason but as the function might change over time, other reasons to fail might get added.

Reply:
Makes sense. fixed